activerecord/test/cases/integration_test.rb

```diff
@@ -29,10 +30,23 @@ def test_to_param_class_method
     assert_equal '4-flamboyant-software', firm.to_param
   end
 
+  def test_to_param_class_method_truncates_words_properly
+    firm = Firm.find(4)
+    firm.name << ', Inc.'
+    assert_equal '4-flamboyant-software', firm.to_param
+  end
+
+  def test_to_param_class_method_truncates_after_parameterize
+    firm = Firm.find(4)
+    firm.name = "Huey, Dewey, & Louie LLC"
```

Comment:
I don't understand your concern. The benefit to parameterizing prior to truncation is that characters that will no make it into the final result do not influence the truncation. In this case, the commas and the ampersand will not be used in the result, but that makes the length 4 characters more for no reason. In this way we are truncating the URL parameter rather than parameterizing the truncated attribute.